### ISSUE TITLE
Compiler errors when building latest code with build-bigtable.sh

### ISSUE THREAD

**Issue by itzg** [OpenTSDB/opentsdb#1959]
Due to this issue https://github.com/OpenTSDB/opentsdb/issues/1899 I wanted to build a bigtable variant of opentsdb using `build-bigtable.sh` from the latest on master (12bc21ddcc3d559c8d1ca696906888a88d056876) in order to pick up this PR https://github.com/OpenTSDB/opentsdb/pull/1903; however, the build fails due to compiler errors:

```
4 errors
100 warnings
make[1]: Leaving directory '/build/build'
make[1]: *** [Makefile:2012: .javac-unittests-stamp] Error 1
make: *** [Makefile:1401: all] Error 2
```
where the full build logs are [here](https://gist.github.com/itzg/0af086ee4389c2dda88fe2ef0cf9cc47#file-gistfile1-txt).

I even tried `build.sh` but get a different error count:
```
1 error
100 warnings
make[1]: *** [Makefile:2012: .javac-unittests-stamp] Error 1
make[1]: Leaving directory '/build/build'
make: *** [Makefile:1401: all] Error 2
```

where the full build logs are [here](https://gist.github.com/itzg/582b49647b310d50435570899097dd8e#file-gistfile1-txt) 

In case it helps, here is the Dockerfile I temporarily created to do the `build-bigtable.sh` build:
```dockerfile
FROM openjdk:8-jdk as builder

RUN apt-get update && \
    DEBIAN_FRONTEND=noninteractive apt-get install -y \
      make \
      autoconf \
      python \
      gnuplot \
    && apt-get clean

WORKDIR /build
COPY . .

RUN bash build-bigtable.sh
```

**Comment by rgarg-quotient** [comment]
I am getting the same issue with same error. Error is as below:

`./test/core/TestTsdbQueryQueries.java:1609:` error: incompatible types: ArrayList<TagVLiteralOrFilter> cannot be converted to List<TagVFilter>
    sub.setFilters(Lists.newArrayList(new TagVLiteralOrFilter("host", TAGV_STRING)));`

**Comment by pansingh-qt** [comment]
yes, i am also getting same error while running the build-bigtable.sh and scripts fails with following:

`    sub2.setTags(tags);
        ^
../test/core/TestTSSubQuery.java:385: warning: [deprecation] setTags(Map<String,String>) in TSSubQuery has been deprecated
    sub1.setTags(tags);
        ^
../test/core/TestTSSubQuery.java:395: warning: [deprecation] setTags(Map<String,String>) in TSSubQuery has been deprecated
    sub2.setTags(tags);
        ^
Note: Some input files additionally use or override a deprecated API.
Note: ../test/stats/TestQueryStats.java uses unchecked or unsafe operations.
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error
100 warnings
make[1]: *** [Makefile:2012: .javac-unittests-stamp] Error 1
make: *** [Makefile:1401: all] Error 2`

On further looking, found that in between these error logs, there is one piece which shows error as below:
`../test/core/TestTsdbQueryQueries.java:1609: error: incompatible types: ArrayList<TagVLiteralOrFilter> cannot be converted to List<TagVFilter>
    sub.setFilters(Lists.newArrayList(new TagVLiteralOrFilter("host", TAGV_STRING)));`

Repo: https://github.com/OpenTSDB/opentsdb 
Branch: master

**Comment by Rajat-Perf** [comment]
I am also getting same error while building latest code with build.sh . 
Below is the error count :- 

1 error
100 warnings
Makefile:2009: recipe for target '.javac-unittests-stamp' failed
make[1]: *** [.javac-unittests-stamp] Error 1
make[1]: Leaving directory '/opt/opentsdb/build'
Makefile:1401: recipe for target 'all' failed
make: *** [all] Error 2

Full logs are here
[build_sh_logs.txt](https://github.com/OpenTSDB/opentsdb/files/4981785/build_sh_logs.txt)
Please help me to resolve this issue .

**Comment by ugtar** [comment]
I don't know if it's the java version I'm using, but I had to rebase and remove four recent commits to get a successful compilation. I removed:
```
81d21126 ExplicitTags filtering with FuzzyFilters
e8f49005 Fix PR 1896 with the fuzzy filter list so that it will honor...

1d3b180b Test rollup filter fix for #1083
12bc21dd  Tweak TestTsdbQueryQueries to pass in older java versions
```
I'm not sure what these are for. The last two are a test for functionality which was added, as well as a fix for that test

**Comment by manolama** [comment]
It was a tweak to the AsyncHBase API that wasn't ported to the Bigtable implementation. I just cut a release of the Bigtable client with https://github.com/OpenTSDB/asyncbigtable/commit/ab279eece3f4c94887237497626d585471c02e3d and as soon as it's public in Sonatype I'll update the version here.
